src/migration/1697624128431-TagRecommendation.ts

```diff
@@ -0,0 +1,49 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class TagRecommendation1697624128431 implements MigrationInterface {
+    name = 'TagRecommendation1697624128431'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const isNonProductionEnv = ['development', 'test'].includes(process.env.NODE_ENV)
+        const createdAtThreshold = isNonProductionEnv ? `'1970-01-01'` : `(current_timestamp - interval '60 day')::date`;
+        const totalThreshold = isNonProductionEnv ? 1 : 100;
```

Comment:
This is here because in local environment the data conditions can not fulfil thresholds we use in production. This makes sure when we run mutations in development/test we can still have the recommendations view to return something.

Open for better alternatives if anybody has it.

Reply:
I don't think we need it for production, we can just call the refresh function. For local maybe we can utilize seeds? We don't need anything fancy or right, we just need something

Reply:
Not sure what do you mean by not needing it in production? This migration sets up the view for usage. 

As for seeding data locally, we are trying to solve two issues:

> It requires for keyword to be used in more then 100 posts

We could solve it with seeding but we would need more posts and more keywords spread out. I guess we would need to pull some slice of production data into seeds to do it quickly. With lowering the total threshold we will naturally get recommendations from our seed data (I checked it works quite good),

> In production we are showing posts created in last 60 days

With seed posts are created few years ago, and also if we do add dynamic `createdAt` to seed it would again expire in 60 days.

My logic with above is that since we always reset our local env multiple times in a week it was better to do this adjustment to view threshold locally to avoid adding more data or having a dynamic createdAt which would again never satisfy constraints. 

In production this will run once and function as any other migration.

Reply:
Ignore me, I thought it's for something else. Let's keep it as is for now. It's one off